### ISSUE TITLE
Depth estimation on invalid projection area

### ISSUE THREAD

**Issue by Hubert3123** [Syniez/SlaBins#4]
Thanks for the great work and the release of augmented data and corresponding code. I might want to ask how do you guys make depth estimation on areas on the fisheye images that are, for example, self-occluded (remain static to the fisheye view) or do not have a valid projection of photometric information (black areas) due to distortion or projection due to augmentation? Meanwhile, I also kind of wonder if the augmentation of viewing direction would change camera intrinsics ? (I cannot come up with a reason why it would change it, but still being a bit curious). 
Thanks! Looking forward to your reply.

**Comment by Syniez** [comment]
Thanks for your interest in our work!

First, for the missing regions (_black areas_) on the fisheye image, we do not give supervision during the training step. We mask black areas according to the valid pixel region (_regions that have ground-truth depth value_) when we measure the loss. So, on the black regions depth loss is actually not measured, but the model predicts similar values considering neighborhood pixels.

Next, our viewing direction augmentation doesn't change the camera intrinsic parameters. It only changes the camera's extrinsic parameters especially, rotation metrics.

Thank you.

**Comment by Hubert3123** [comment]
> Thanks for your interest in our work!
> 
> First, for the missing regions (_black areas_) on the fisheye image, we do not give supervision during the training step. We mask black areas according to the valid pixel region (_regions that have ground-truth depth value_) when we measure the loss. So, on the black regions depth loss is actually not measured, but the model predicts similar values considering neighborhood pixels.
> 
> Next, our viewing direction augmentation doesn't change the camera intrinsic parameters. It only changes the camera's extrinsic parameters especially, rotation metrics.
> 
> Thank you.

Thanks so much for the instant answer! Is it possible that you could also share the valid masks on each augmented fisheye views?

**Comment by Syniez** [comment]
It would be better for you to write codes for making a valid depth mask rather than download all corresponding valid masks. Following the conventions, we make a valid mask from the ground-truth depth map within a pre-defined valid depth range as follows:

`valid_mask = np.logical_and(gt_depth > args.min_depth_eval, gt_depth < args.max_depth_eval)`

You can use this code snippet in your dataloader or trainer file.

Thank you.

**Comment by booker-max** [comment]
> It would be better for you to write codes for making a valid depth mask rather than download all corresponding valid masks. Following the conventions, we make a valid mask from the ground-truth depth map within a pre-defined valid depth range as follows:
> 
> `valid_mask = np.logical_and(gt_depth > args.min_depth_eval, gt_depth < args.max_depth_eval)`
> 
> You can use this code snippet in your dataloader or trainer file.
> 
> Thank you.

"Could you please advise on the appropriate values for min_depth_eval and max_depth_eval when working with the KITTI360 and SynWoodScape datasets you have processed?"

**Comment by booker-max** [comment]
> > It would be better for you to write codes for making a valid depth mask rather than download all corresponding valid masks. Following the conventions, we make a valid mask from the ground-truth depth map within a pre-defined valid depth range as follows:
> > `valid_mask = np.logical_and(gt_depth > args.min_depth_eval, gt_depth < args.max_depth_eval)`
> > You can use this code snippet in your dataloader or trainer file.
> > Thank you.
> 
> "Could you please advise on the appropriate values for min_depth_eval and max_depth_eval when working with the KITTI360 and SynWoodScape datasets you have processed?"

I got. The paper mentions that the maximum depth is 40.0 m. Thank you
